src/internal/analytics/components/analytics-funnel.tsx

```diff
@@ -180,7 +180,7 @@ export const AnalyticsFunnelStep = ({ children, stepNumber, stepNameSelector }:
 
     return () => {
       //eslint-disable-next-line react-hooks/exhaustive-deps
-      if (funnelInteractionId && funnelState.current === 'default') {
+      if (funnelInteractionId && funnelState.current !== 'cancelled') {
```

Comment:
are there any other states besides `default` and `cancelled`?

Reply:
Yes, the possible values are defined in this type: https://github.com/cloudscape-design/components/blob/eda050146b48c90e6cb47b7037832367e9047300/src/internal/analytics/context/analytics-context.ts#L6

The `validating` state is similar to the `default` state, except that the funnel is in a loading state (detected thought the primary button having `loading={true}`). If the step unmounts during either of these two states (but the funnel hasn't been cancelled), then the step was successful, and should emit the event here.

The `complete` state is entered when the funnel has been successfully submitted, i.e. the resource has been created. We still want to emit `funnelStepComplete` when the step unmounts, so that the step has a start and end that can later be used in calculations.